CHANGES.rst

```diff
@@ -27,6 +27,9 @@ Version 3.1.0 (2019-XX-XX)
 - fix issue where it was possible to add an index to an existing dataset by using update functions and partition indices
   (https://github.com/JDASoftwareGroup/kartothek/issues/16).
 
+- fix issue where unreferenced files were not being removed when deleting an entire dataset and create test
```

Comment:
Don't mention the test in the changelog. We're always writing tests and the user is not interested in this